src/loaders/svg/tvgSvgLoader.cpp

```diff
@@ -2213,7 +2216,12 @@ static void _svgLoaderParserXmlOpen(SvgLoaderData* loader, const char* content,
         stop->a = 255;
         simpleXmlParseAttributes(attrs, attrsLength, _attrParseStops, loader);
         if (loader->latestGradient) {
+            // check the offset corner cases - refer to: https://svgwg.org/svg2-draft/pservers.html#StopNotes
+            if (stop->offset < loader->latestGradient->prevOffset) stop->offset = loader->latestGradient->prevOffset;
+            else if (stop->offset > 1) stop->offset = 1;
```

Comment:
I don't think it's a good idea to modify the parsed data.
There are some case what parsed data is modified. But basically I think we should keep the parsed data.
When we create a tvg object using the parsed data, we can implement it according to the spec we think.

I mean...
You can apply your idea when creating colorStops of fill->Grad in In _applyLinearGradientProperty().

like this...?
```c
71     //Update the stops     
72     stopCount = g->stops.count;
73     if (stopCount > 0) {       
74         stops = (Fill::ColorStop*)calloc(stopCount, sizeof(Fill::ColorStop));
75         if (!stops) return fillGrad;      
76         float prevOffset = 0.0;           
77         for (uint32_t i = 0; i < g->stops.count; ++i) {
78             auto colorStop = g->stops.data[i];         
79             //Use premultiplied color
80             stops[i].r = colorStop->r;                            
81             stops[i].g = colorStop->g;
82             stops[i].b = colorStop->b;
83             stops[i].a = (colorStop->a * fillOpacity) / 255.0f;
84             stops[i].offset = colorStop->offset; 
85             if (colorStop->offset < prevOffset) stops[i].offset = prevOffset;
86             else if (colorStop->offset > 1) stops[i].offset = 1;
87             prevOffset = colorStop->offset;     
88         }
89         fillGrad->colorStops(stops, stopCount);
90         free(stops); 

```

Reply:
yes, you've told me already once not to change the parsed data, I forgot. I'll remember in the future :)